core/unix/signal_linux_x86.c

```diff
@@ -445,7 +445,7 @@ sigcontext_to_mcontext_simd(priv_mcontext_t *mc, sig_full_cxt_t *sc_full)
     sigcontext_t *sc = sc_full->sc;
     if (sc->fpstate != NULL) {
         int i;
-        for (i = 0; i < NUM_SIMD_SLOTS; i++) {
+        for (i = 0; i < proc_num_simd_saved(); i++) {
```

Comment:
Are you sure about these cases where you did s/NUM_SIMD_SLOTS/proc_num_simd_saved()/?  You've done that for the sigcontext and CONTEXT cases.  However, I am not sure it's correct: for the Windows cases where only 6 are saved normally, when we do an absolute context switch I think we do need all 16.   Consider the app calling SetThreadContext: we want all 16.

It seems a little safer for avx512 being unused up to that point: but again what if an absolute context wants to set avx512 state b/c *future* code is going to use it?

I think all these cases where we deal with *absolute* contexts need to use all slots: the optimized case of only the "saved" ones seems to only apply for *relative* contexts.  I guess that means we have to pay the frequency scaling here: but these kernel-mediated events are already expensive.

Reply:
As discussed, I will NUM_SIMD_SAVED => proc_num_simd_saved() and NUM_SIMD_SLOTS => MCXT_NUM_SIMD_SLOTS.